plugin/trino-druid/src/test/java/io/trino/plugin/druid/BaseDruidConnectorTest.java

```diff
@@ -323,4 +325,189 @@ public void testNativeQueryInsertStatementTableExists()
+                .matches("VALUES (BIGINT '3', BIGINT '1673', CAST('RAIL' AS varchar))")
+                .isFullyPushedDown();
+
+        assertThat(query("SELECT linenumber, partkey, shipmode FROM lineitem WHERE __time <= TIMESTAMP '1992-01-04'"))
+                .matches("VALUES (BIGINT '3', BIGINT '1673', CAST('RAIL' AS varchar))")
+                .isFullyPushedDown();
+
+        assertThat(query("SELECT linenumber, partkey, shipmode FROM lineitem WHERE __time > TIMESTAMP '1998-11-28'"))
+                .matches("VALUES " +
+                        "(BIGINT '2', BIGINT '370', CAST('RAIL' AS varchar)), " +
+                        "(BIGINT '2', BIGINT '468', CAST('AIR' AS varchar))")
+                .isFullyPushedDown();
+
+        assertThat(query("SELECT linenumber, partkey, shipmode FROM lineitem WHERE __time >= TIMESTAMP '1998-11-29 00:00:00'"))
+                .matches("VALUES " +
+                        "(BIGINT '2', BIGINT '370', CAST('RAIL' AS varchar)), " +
+                        "(BIGINT '2', BIGINT '468', CAST('AIR' AS varchar))")
+                .isFullyPushedDown();
+
+        // timestamp range
+        assertThat(query("SELECT linenumber, partkey, shipmode FROM lineitem WHERE __time BETWEEN TIMESTAMP '1992-01-01' AND TIMESTAMP '1992-01-05'"))
+                .matches("VALUES (BIGINT '3', BIGINT '1673', CAST('RAIL' AS varchar))")
+                .isFullyPushedDown();
+
+        // varchar IN without domain compaction
+        assertThat(query("SELECT linenumber, partkey, shipmode FROM lineitem WHERE __time IN (TIMESTAMP '1992-01-04', TIMESTAMP '1998-11-27 00:00:00.000', TIMESTAMP '1998-11-28')"))
+                .matches("VALUES " +
+                        "(BIGINT '3', BIGINT '1673', CAST('RAIL' AS varchar)), " +
+                        "(BIGINT '1', BIGINT '574', CAST('AIR' AS varchar))")
+                .isFullyPushedDown();
+
+        // varchar IN with small compaction threshold
+        assertThat(query(
+                Session.builder(getSession())
+                        .setCatalogSessionProperty("druid", "domain_compaction_threshold", "1")
+                        .build(),
+                "SELECT linenumber, partkey, shipmode FROM lineitem WHERE __time IN (TIMESTAMP '1992-01-04', TIMESTAMP '1998-11-27 00:00:00', TIMESTAMP '1998-11-28')"))
+                .matches("VALUES " +
+                        "(BIGINT '3', BIGINT '1673', CAST('RAIL' AS varchar)), " +
+                        "(BIGINT '1', BIGINT '574', CAST('AIR' AS varchar))")
+                // Filter node is retained as no constraint is pushed into connector.
+                .isNotFullyPushedDown(FilterNode.class);
+    }
+
+    @Test
+    public void testPredicatePushdownForTimestampWithMillisPrecision()
+    {
+        // timestamp equality
+        assertThat(query("SELECT linenumber, partkey, shipmode FROM lineitem WHERE __time = TIMESTAMP '1992-01-04 00:00:00.001'"))
```

Comment:
predicates with precision other than 3 would be useful to test as well

Reply:
Added tests with higher precision.